### PR TITLE
fix(deps): update to latest hapi, joi and boom. requires node 4+

### PR DESCRIPTION
**Pull request by vladikoff**
Fixes https://github.com/mozilla/fxa-profile-server/issues/209

### CODE REVIEW

File: Dockerfile-build [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM node:0.10.46
+FROM node:4.4.7
```
Comment (vladikoff) [inline Dockerfile-build]: @jbuck is 4.4.7 here okay?

Comment (jbuck) [inline Dockerfile-build]: @vladikoff 👍

**Comment by seanmonstar** [comment]
Merge conflicts, what's the blocked for?

**Comment by vladikoff** [comment]
> what's the blocked for?

I'm thinking we might need to wait until we switch to node 4 in prod.

**Comment by jbuck** [comment]
@vladikoff I think it's time to land this PR

**Comment by vladikoff** [comment]
@seanmonstar final r?

**Pull request merged**